### PR TITLE
refactor: move Module and function_type helper to module.hpp

### PR DESCRIPTION
**Pull request by gumb0**
Needed for https://github.com/wasmx/fizzy/pull/267

### CODE REVIEW

File: lib/fizzy/module.hpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#pragma once
```
Comment (chfast) [inline lib/fizzy/module.hpp:1]: Now you need the header comment.

File: lib/fizzy/module.hpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+#include "types.hpp"
+#include <cassert>
+
+namespace fizzy
+{
+inline const FuncType& function_type(const Module& module, FuncIdx idx)
```
Comment (chfast) [inline lib/fizzy/module.hpp:8]:
```suggestion
inline const FuncType& get_function_type(const Module& module, FuncIdx idx) noexcept
```

File: lib/fizzy/module.hpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2019-2020 The Fizzy Authors.
```
Comment (chfast) [inline lib/fizzy/module.hpp:2]:
```suggestion
// Copyright 2020 The Fizzy Authors.
```

File: lib/fizzy/module.hpp [hunk 4]
```diff
@@ -9,7 +9,7 @@
 
 namespace fizzy
 {
-inline const FuncType& function_type(const Module& module, FuncIdx idx)
+inline const FuncType& get_function_type(const Module& module, FuncIdx idx)
```
Comment (chfast) [inline lib/fizzy/module.hpp:49]:
```suggestion
inline const FuncType& get_function_type(const Module& module, FuncIdx idx) noexcept
```

Comment (chfast) [reply lib/fizzy/module.hpp:49]: You can also make it a `Module`'s method now, but I'm not pushing this. It's rather blurry line what should be a method and what a free function.

Comment (gumb0) [reply lib/fizzy/module.hpp]: I'll make it a method, should it still be called `get_function_type` ?

Comment (chfast) [reply lib/fizzy/module.hpp]: Yes.

**Pull request merged**